internal/resourcemodifiers/resource_modifiers.go

```diff
@@ -0,0 +1,152 @@
+	GroupKind         string   `yaml:"groupKind"`
+	ResourceNameRegex string   `yaml:"resourceNameRegex"`
+}
+
+type ResourceModifierRule struct {
+	Conditions Conditions  `yaml:"conditions"`
+	Patches    []JSONPatch `yaml:"patches"`
+}
+
+type ResourceModifiers struct {
+	Version               string                 `yaml:"version"`
+	ResourceModifierRules []ResourceModifierRule `yaml:"resourceModifierRules"`
+}
+
+func GetResourceModifiersFromConfig(cm *v1.ConfigMap) (*ResourceModifiers, error) {
+	if cm == nil {
+		return nil, fmt.Errorf("could not parse config from nil configmap")
+	}
+	if len(cm.Data) != 1 {
+		return nil, fmt.Errorf("illegal resource modifiers %s/%s configmap", cm.Name, cm.Namespace)
+	}
+
+	var yamlData string
+	for _, v := range cm.Data {
+		yamlData = v
+	}
+
+	resModifiers, err := unmarshalResourceModifiers(&yamlData)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	return resModifiers, nil
+}
+
+func (p *ResourceModifiers) ApplyResourceModifierRules(obj *unstructured.Unstructured, groupResource string, log logrus.FieldLogger) []error {
+	var errs []error
+	for _, rule := range p.ResourceModifierRules {
+		err := rule.Apply(obj, groupResource, log)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	return errs
+}
+
+func (r *ResourceModifierRule) Apply(obj *unstructured.Unstructured, groupResource string, log logrus.FieldLogger) error {
+	namespaceInclusion := collections.NewIncludesExcludes().Includes(r.Conditions.Namespaces...)
+	if !namespaceInclusion.ShouldInclude(obj.GetNamespace()) {
+		return nil
+	}
+	if !strings.EqualFold(groupResource, r.Conditions.GroupKind) {
+		return nil
+	}
+	if r.Conditions.ResourceNameRegex != "" {
+		match, _ := regexp.MatchString(r.Conditions.ResourceNameRegex, obj.GetName())
+		if !match {
+			return nil
+		}
+	}
+	patches, err := r.PatchArrayToByteArray()
+	if err != nil {
+		return err
+	}
+	err = ApplyPatch(patches, obj, log)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// convert all JsonPatch to string array with the format of jsonpatch.Patch and then convert it to byte array
+func (r *ResourceModifierRule) PatchArrayToByteArray() ([]byte, error) {
+	var patches []string
+	for _, patch := range r.Patches {
+		patches = append(patches, patch.ToString())
+	}
+	patchesStr := strings.Join(patches, ",\n\t")
+	return []byte(fmt.Sprintf(`[%s]`, patchesStr)), nil
+}
+
+func (p *JSONPatch) ToString() string {
+	return fmt.Sprintf(`{"op": "%s", "path": "%s", "value": "%s"}`, p.Operation, p.Path, p.Value)
+}
+
+func ApplyPatch(patch []byte, obj *unstructured.Unstructured, log logrus.FieldLogger) error {
+	jsonPatch, err := jsonpatch.DecodePatch(patch)
+	if err != nil {
+		return fmt.Errorf("error in decoding json patch %s", err.Error())
+	}
+	objBytes, err := obj.MarshalJSON()
+	if err != nil {
+		return fmt.Errorf("error in marshaling object %s", err.Error())
+	}
+	modifiedObjBytes, err := jsonPatch.Apply(objBytes)
```

Comment:
Seems an error is returned if no path matches the `test` operation. Correct me if I'm wrong.
So that means if users set the `test` patch, they must make sure it matches, or the restore will fail.

Reply:
the restore will not fail
just that particular patch would fail and restore will go on as it is.

For example there are 5 PVCs which match the "conditions"

but only 1 PVC which satisfies the "test" operator. Only that PVC will be patched and all the others will get skipped in some sense. 
The logs will show that, but there won't be any hard failure.

Reply:
Yes, the error is shown in the logs, but won't the restore be marked as partially failed? Is this the expected behavior?

Reply:
Handled the "test operator related errors separately to avoid partial failure states based on your feedback. Updated code with test case.